crates/stages-api/src/pipeline/mod.rs

```diff
@@ -66,7 +67,7 @@ pub type PipelineWithResult<DB> = (Pipeline<DB>, Result<ControlFlow, PipelineErr
 /// # Defaults
 ///
 /// The [DefaultStages](crate::sets::DefaultStages) are used to fully sync reth.
-pub struct Pipeline<DB: Database> {
+pub struct Pipeline<DB: Database + Clone> {
     /// Provider factory.
     provider_factory: ProviderFactory<DB>,
```

Comment:
`Pruner` requires an instance of `provider_factory`. Either we make `Database + Clone`, or would need to add an Arc. I'm fine with either.